src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/services/pageService.js

```diff
@@ -21,10 +21,18 @@ const PageService = function () {
         const api = getOverridablePagesApi();
         let request = page;
 
-        request = {
-            ...request,
-            url: ""
-        };
+        if (page.tabId === 0 && !securityService.isSuperUser()) {
+            request = {
+                ...request,
+                parentId: page.parentId !== undefined && page.parentId !== null ? utils.getCurrentPageId() : page.parentId
+            };
+        }
+        else {
+            request = {
+                ...request,
+                url: ""
+            };
+        }
```

Comment:
I think this fragment is kind of verbose. I'd simplify and reduce nesting by extracting the `parentId`/`url` into a variable and inline `page` to have a single assignment

```js
var request = { ...page, ...additionalParams };
```

Reply:
No we don't need any backend changes, just need to give all permissions to the user, whom we want to duplicate the page